File: src/main/java/hms/tecktalks/kotlinvsjava/java/ControlFlows.java
package hms.tecktalks.kotlinvsjava.java;

public class ControlFlows {

    public static void main(String[] args) {
        int max = getMax(40, 50);
        System.out.println("Max : " + max);

        int intVal = getIntVal("One");
        System.out.println("Int val: " + intVal);
    }

    private static int getMax(int a, int b) {
        if (a > b) {
            return a;
        } else {
            return b;
        }
    }

    private static int getIntVal(String name) {
        switch (name) {
            case "One":
                return 1;
            case "Two":
                return 2;
            default:
                return 0;
        }
    }
}

File: src/main/kotlin/hms/tecktalks/kotlinvsjava/kotlin/ControlFlows.kt
package hms.tecktalks.kotlinvsjava.kotlin

fun main(args: Array<String>) {
    val max = getMaxVal(40, 50)
    val max2 = getMaxValV2(40, 50)
    println("Max : $max")
    println("Max2 : $max2")

    val intVal = getIntVal("Two")
    println("Int val $intVal")
}

//Traditional if-else
fun getMaxVal(a: Int, b: Int): Int {
    if (a > b) {
        return a
    } else {
        return b
    }
}

//if-else as an expression
fun getMaxValV2(a: Int, b: Int): Int {
    val max = if (a > b) {
        a
    } else {
        b
    }

    return max
}

fun getIntVal(str: String): Int {
    return when (str) {
        "One" -> 1
        "Two" -> 2
        else -> 0
    }
}